src/plugins/aws/ssh.ts

```diff
@@ -112,11 +112,12 @@ export const awsSshProvider: SshProvider<
 
   requestToSsh: (request) => {
     const { permission, generated } = request;
-    const { awsResourcePermission, instanceId, accountId, region } =
-      permission.spec;
-    const { idcId, idcRegion } = awsResourcePermission.permission;
-    const { ssh, name } = generated;
-    const { linuxUserName } = ssh;
+    const { resource, region } = permission;
+    const { idcId, idcRegion, instanceId, accountId } = resource;
+    const {
+      linuxUserName,
+      resource: { name },
+    } = generated;
```

Comment:
Nested destructuring gets a bit verbose and a bit complicated:

```suggestion
    const { linuxUserName, resource } = generated;
    const { name } = resource;
```

Reply:
There's a collision between `permission.resource` and `generated.resource` but i used an alias to make it work 👍